compiler/luci-interpreter/src/kernels/Maximum.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ * Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "kernels/Maximum.h"
+
+#include "kernels/Utils.h"
+
+#include "kernels/BinaryOpCommon.h"
+
+namespace luci_interpreter
+{
+namespace kernels
+{
+
+Maximum::Maximum(const Tensor *input1, const Tensor *input2, Tensor *output)
+    : Kernel({input1, input2}, {output})
+{
+}
+
+void Maximum::configure()
+{
+  LUCI_INTERPRETER_CHECK(input1()->element_type() == input2()->element_type())
+  LUCI_INTERPRETER_CHECK(input1()->element_type() == output()->element_type())
+  output()->resize(calculateShapeForBroadcast(input1()->shape(), input2()->shape()));
+}
+
+void Maximum::execute() const
+{
+  switch (input1()->element_type())
+  {
+    case DataType::FLOAT32:
+      evalFloat();
+      break;
+    case DataType::U8:
+      evalQuantized();
+      break;
+    default:
+      throw std::runtime_error("Unsupported type.");
+  }
+}
+
+void Maximum::evalFloat() const
+{
+  BinaryOpBroadcastSlow(getTensorShape(input1()), getTensorData<float>(input1()),
+                        getTensorShape(input2()), getTensorData<float>(input2()),
+                        getTensorShape(output()), getTensorData<float>(output()),
+                        [](float x, float y) { return std::max(x, y); });
+}
+
+void Maximum::evalQuantized() const
+{
+  BinaryOpBroadcastSlow(getTensorShape(input1()), getTensorData<uint8_t>(input1()),
+                        getTensorShape(input2()), getTensorData<uint8_t>(input2()),
+                        getTensorShape(output()), getTensorData<uint8_t>(output()),
+                        [](uint8_t x, uint8_t y) { return std::max(x, y); });
+}
```

Comment:
I think you can just in-line these two methods into `execute`, as they are very small.

Reply:
I think in-lining these code could harm the source uniformity (since other kernel sources keep this format)
Don't you think it's better to keep this format?

I will follow your decision.
Thank you.

Reply:
> other kernel sources keep this format

Most of them are, but that's not necessary and certainly not a rule (see DepthToSpace, Squeeze, Unpack and some others). We usually split `execute` method into several `eval*` methods when keeping everything in `execute` method would make it too complex to read / understand. This one looks very simple -- just one function call; I would not make separate methods just for a function call :)

NB
Uniformity is important, but it also has downside -- it is hard to keep it. Once you changed one file, you will have to look through all others and check if they need to be changed too.
Things get worse when you work in parallel with other developers: you might have finally changed all the files and created a PR, but at the same time there could be parallel PR which did not take your changes into account. After you both merged, there will be inconsistency again.

Reply:
> NB
> Uniformity is important, but it also has downside -- it is hard to keep it. Once you changed one file, you will have to look through all others and check if they need to be changed too.
> Things get worse when you work in parallel with other developers: you might have finally changed all the files and created a PR, but at the same time there could be parallel PR which did not take your changes into account. After you both merged, there will be inconsistency again.

Thank you for your sincere and kind advice, @s-barannikov ,
It is meaningful to me to have conversations like this with you :)

I will put that in my mind when I develop with other people.
Thank you so much! :)